pkg/exchange/binance/exchange.go

```diff
@@ -48,6 +49,47 @@ func (e *Exchange) Name() types.ExchangeName {
 	return types.ExchangeBinance
 }
 
+func (e *Exchange) QueryTickers(ctx context.Context, symbol ...string) (map[string]types.Ticker, error) {
+	var ret = make(map[string]types.Ticker)
+	listPriceChangeStatsService := e.Client.NewListPriceChangeStatsService()
+
+	if len(symbol) == 1 {
+		listPriceChangeStatsService.Symbol(symbol[0])
+	}
+
+	changeStats, err := listPriceChangeStatsService.Do(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	m := make(map[string]bool)
+
+	for _, s := range symbol {
+		m[strings.ToUpper(s)] = true
+	}
+
+	for _, retrievedStats := range changeStats {
+		if _, ok := m[retrievedStats.Symbol]; len(symbol) != 0 && !ok {
```

Comment:
We test `len(symbol) != 0` because
* if caller does not pass any symbol, we request all available symbols from Binance for them; all tickers will be in `ret` (= will not "throw away" any tickers)
* if caller passes one symbol, we only request that symbol from Binance; it will be copied to `ret` (= will not throw away any tickers)
* if caller passes more than one symbol, we actually query all symbols from Binance; but only return what caller specified in `ret` (= some of the tickers might be thrown away)

Without this `if` block, we will be constructing a map that's larger than the caller needs 😆

Reply:
oh yes that makes sense